pop11_grammar_ebnf.txt

```diff
@@ -29,7 +34,7 @@ Variable ::=
 	| 'nonsyntax' SyntaxWord
 
 Declaration ::=
-	('global'? ('vars' | 'constant') Varslist | 'lvars' (',' | Word)+) ';'
+	('global'? ('vars' 'procedure'? | 'constant') Varslist | ('lvars' | 'lconstant' | 'dlocal') (',' | Word)+) ';'

```

Comment:
The situation with `procedure` is complicated. I suggest leaving this out for a little while. If you want to tackle it then the details are:
 - `vars`, `lvars`, `lconstant` allow the `procedure` modifier to appear ahead of the identifier in the Varslist
 - `dlocal` does not allow the `procedure` modifier.
But `lconstant` requires an initial value to be supplied. 

Since `lconstant` was introduced to the language after this book was written, it is not surprising that it doesn't fit the grammar at all. For the moment I would suggest leaving it out.

Reply:
Sure, I have changed it to the following version. Hopefully it is better now.
```
Declaration ::=
	'global'? ('vars' 'procedure'? | 'constant') Varslist ';'
	| 'lvars' 'procedure'? (',' | Word)+ ';'
	| 'dlocal' (',' | Word)+ ';'
```

Reply:
Close. Consider this code (which I just compiled):
```
 vars procedure a, procedure b;
 
 define foo( procedure p, procedure q );
 enddefine;
```

I think it make more sense to put the `procedure` part into `Varlist`.
```
Varslist ::= ( ','* ( 'procedure' | 'macro' | 'syntax' | Number )? Word )+
```
and remove it from `Declaration`:
```
Declaration ::=
	'global'? ('vars' | 'constant') Varslist ';'
	| 'lvars' (',' | Word)+ ';'
	| 'dlocal' (',' | Word)+ ';'
```

I think that is as good as we will get until we distinguish the two different uses of `Varlist`. The book grammar is significantly wrong here.

Reply:
Done